proposal.rst

```diff
@@ -239,7 +301,82 @@ Project Timeline
 .. any university classes or exams, vacations, etc), make sure you include them
 .. in your project timeline.
 
-FILLME
+MONTH 1  : 
+
+BIG TASK NAME : DOCUMENTATION STEP
+
+Week 1 
+	- Read and understand code-base further. 
+	- Create a new VagrantFile and make sure that it installs all the xapian dependencies. I have a doc where I have all commands which helped me with my Xapian setup.
+	- Update the "Installation" page with vagrant walk through tutorial.
+
+Week 2 
+	- Re-work on "Practical Example" tutorial to make it more user-friendly. 
+	- Add Omega documentation (in Xapian docs) for indexing, and how to obtain a front-end html(omindex, omega "query" [arguments])
+	- Start working on GSOC task:  "Rework Omega templates to use more modern web techniques"
+
+Week 3 
+	- Continue working on Omega templates.
+	- Identify areas that requires change user-end APIs, consult with mentors about this.
+	- Continue working on the user-end APIs.
+
+Week 4
+	- Continue working on the user-end APIs.
+	- Add tests to see consistency of new API(). 
```

Comment:
What sort of tests will highlight consistency problems with the API? Do you have examples (from Xapian or elsewhere) of the sorts of things you're looking for?

Reply:
I'm not sure I understand what wrapping `xapian-docsprint/code/python` achieves.  That seems like it would just obscure the code in there which actually does the work, and that code exists solely as examples to demonstrate how to use particular API features, so clarity is a high priority.

Elsewhere you say about this:
> A user-end API would mean that I call this functionality something like xapian-docsprint --index [DATAFILE], or other flags such as --search, --delete,--search etc.

And how is that useful?

Reply:
This is my old proposal. I have made significant changes to my latest proposal where I concentrated more on reported Systems rather than these documentations.

Reply:
OK - it showed as a new comment from you after you asked me to look at your proposal again.